Lib/fontmake/font_project.py

```diff
@@ -420,6 +450,33 @@ def _output_path(self, ufo, ext, is_instance=False, interpolatable=False,
             os.makedirs(out_dir)
         return os.path.join(out_dir, '%s.%s' % (self._font_name(ufo), ext))
 
+    def _designspace_locations(self, designspace_path):
+        """Map font filenames to their locations in a designspace."""
+
+        maps = []
+        for location_list in varLib.designspace.load(designspace_path):
+            location_map = {}
+            for loc in location_list:
+                abspath = os.path.normpath(os.path.join(
+                    os.path.dirname(designspace_path), loc['filename']))
+                location_map[abspath] = loc['location']
+            maps.append(location_map)
+        return maps
+
+    def _closest_location(self, location_map, target):
+        """Return path of font whose location is closest to target."""
+
+        dist = lambda a, b: math.sqrt(sum((a[k] - b[k]) ** 2 for k in a.keys()))
```

Comment:
I'm thinking, say axis A runs on a scale from 1-100, and B on a scale from 1-10.  A difference in A of 10 points is semantically 'equivalent' to a difference in B of 1 point, but the distance in A looks a lot larger.  This function will choose fonts with almost any B value over those with even small differences in A. So I figure these must be scaled so that differences on the various axes all contribute the 'intended' amount to the selection algorithm, even if the ranges are not exactly the same they should be close.

I guess also there are only two or maybe three axes?